client/src/client.rs

```diff
@@ -126,40 +126,36 @@ impl Client {
             None,
         );
 
-        let mut response = server_api
+        let response = server_api
             .register_client(phone_number, password.to_owned(), req, None)
             .await
             .map_err(|_| RegistrationError::NoResponse)?;
```

Comment:
Why does this not handle the errors which was previously handled by the match case?